deployer/orb.yaml

```diff
@@ -1,6 +1,85 @@
 version: 2.1
 description: "Tools for running deployment commands"
 
+commands:
+  validate-tag-for-pip:
+    description: >
+      Uses pips own Version parser to ensure semver semantics of the tag match
+      exactly with those used by pip.
+    parameters:
+      verify_version:
+        default: CIRCLE_TAG
+        description: >
+          The string or string-name of env var which holds the version string
+          to parse with pip Version
+        type: string
+      should_upgrade_pip:
+        type: boolean
+        default: false
+    steps:
+      - when:
+          condition: <<parameters.should_upgrade_pip>>
+          steps:
+            - run: python3 -m pip install --upgrade pip
+      - run:
+          name: Validate tag format for pip
+          shell: /usr/local/bin/python
+          command: |
+            import os
+            from pip._vendor.packaging.version import Version
+            try:
+              # See if verify_version is env var name holding version value
+              tag = os.environ['<<parameters.verify_version>>']
+            except KeyError:
+              # Check if verify_version itself was a version string
+              tag = '<<parameters.verify_version>>'
+            v = Version(tag)
+            assert tag == str(v), f'Original tag ({tag}) should have been {str(v)}'
+            assert v.is_devrelease or not v.local, f'tag should not contain \'+{v.local}\' unless dev version'
+
+  verify-tag-matches-deploy:
```

Comment:
This and `upload` are both overly specific names for a "deployer" orb. Perhaps `python-sdist-verify-tag-matches-deploy` and `upload-to-gemfury`?